### PR TITLE
Add functionality of server variables for the OAS plugin

### PR DESCRIPTION
**Pull request by hamzamihaidanielx**
```markdown
## 🎯 Changes

Adds a new `serverVariables` option to `@kubb/plugin-oas` that resolves OpenAPI server URL template variables (e.g. `{env}`, `{port}`) when computing the base URL.

Currently, `serverIndex` returns the raw server URL string including unresolved `{variable}` placeholders. This means specs with templated server URLs like:

```yaml
servers:
  - url: https://api.{env}.example.com
    variables:
      env:
        default: dev
        enum: [dev, staging, prod]
```

produce a broken base URL (`https://api.{env}.example.com`) that downstream plugins (`plugin-client`, `plugin-react-query`) embed as-is into generated code.

With this change, users can provide variable overrides that get substituted into the URL, with unspecified variables falling back to their spec-defined `default` values:

```ts
pluginOas({
  serverIndex: 0,
  serverVariables: { env: 'prod' },
})
// -> baseURL: "https://api.prod.example.com"
```

Enum validation is enforced. Providing a value not in the variable's `enum` list throws a descriptive error at generation time.

**Files changed:**
- `packages/plugin-oas/src/types.ts` - Added `serverVariables?: Record<string, string>` option with JSDoc
- `packages/plugin-oas/src/plugin.ts` - Updated `getBaseURL()` to resolve variables via new utility
- `packages/plugin-oas/src/utils/resolveServerUrl.ts` - New pure utility function for URL variable substitution
- `packages/plugin-oas/src/utils/resolveServerUrl.test.ts` - 9 test cases covering defaults, overrides, multiple variables, enum validation, and edge cases
- `packages/plugin-oas/src/utils/index.ts` - Exported new utility

## ✅ Checklist

- [x] I have followed the steps in the [Contributing guide](https://github.com/kubb-labs/kubb/blob/main/CONTRIBUTING.md).
- [x] I have tested this code locally with `pnpm run test`.

## 🚀 Release Impact

- [x] This change affects published code, and I have generated a [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).
- [ ] This change is for the docs (no release).
```

### CODE REVIEW

**Comment by stijnvanhulle** [comment]
@all-contributors please add @hamzamihaidanielx for code

File: packages/plugin-oas/src/types.ts [hunk 1]
```diff
@@ -50,6 +50,33 @@ export type Options = {
    * - `1` returns `http://localhost:3000`
    */
   serverIndex?: number
+  /**
```
Comment (stijnvanhulle) [inline packages/plugin-oas/src/types.ts:53]: @hamzamihaidanielx Could you also add this to the docs, see `docs/plugins/plugin-oas.md`

Comment (hamzamihaidanielx) [reply packages/plugin-oas/src/types.ts:53]: @stijnvanhulle sure thing, done.

**Comment by hamzamihaidanielx** [comment]
@stijnvanhulle Everything seems fine now but it seems like I still cannot merge the one, we need one more approval. Thank you.

**Comment by stijnvanhulle** [comment]
@hamzamihaidanielx looks good to me, triggers the ci again so you can merge in a couple of minutes

**Comment by hamzamihaidanielx** [comment]
@stijnvanhulle all good not it seems, but I do not have the Merge button visible on my side, I guess you are the only one who can do that. Thanks.
